homeassistant/components/google_travel_time/config_flow.py

```diff
@@ -204,15 +220,44 @@ async def async_step_user(self, user_input=None) -> ConfigFlowResult:
 
         return self.async_show_form(
             step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(
-                        CONF_NAME, default=user_input.get(CONF_NAME, DEFAULT_NAME)
-                    ): cv.string,
-                    vol.Required(CONF_API_KEY): cv.string,
-                    vol.Required(CONF_DESTINATION): cv.string,
-                    vol.Required(CONF_ORIGIN): cv.string,
-                }
+            data_schema=self.add_suggested_values_to_schema(CONFIG_SCHEMA, user_input),
+            errors=errors,
+        )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration."""
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert entry
+
+        errors = {}
+        user_input = user_input or {}
+        if user_input:
+            try:
+                await self.hass.async_add_executor_job(
+                    validate_config_entry,
+                    self.hass,
+                    user_input[CONF_API_KEY],
+                    user_input[CONF_ORIGIN],
+                    user_input[CONF_DESTINATION],
+                )
+                return self.async_update_reload_and_abort(
```

Comment:
Move outside try-block. Only things that can raise should be inside.

Reply:
If we do that we will have to introduce an `if` statement to check for errors. I prefer the current style for readability. If you insist I will change it

Reply:
You should be able to use `else` at the end of the try/except block to execute code if there's been no exception.